### ISSUE TITLE
Aeson Number now uses scientific

### ISSUE THREAD

**Issue by Lingyis** [nh2/aesonbson#3]
As of Aeson 0.7, the scientific package is now a dependency.  So Attoparsec's Number is no longer used, but need to do something like:

```
bsonifyValue (Number n) = case base10Exponent n == 0 of { True -> Int64 $ fromIntegral $ coefficient n
                                                        ; False -> Float $ (fromRational . toRational) n }
```

The code snippet above isn't quite right, however, because the `scientific` does not guarantee that if base10Exponent is 0 then it's an integer, so more careful conversion is needed.

**Comment by cdupont** [comment]
Any update on this?

**Comment by puffnfresh** [comment]
@cdupont we use Nix to specify using this particular commit at work:

https://github.com/puffnfresh/aesonbson/commit/93b4714965fc797e133dd35bce85b7af3a4a4633

It builds up on a few of the forks I found.

**Comment by nh2** [comment]
Fixed in #11, and release `0.4.0` on Hackage.
